controlplane/internal/imageregistry/image_registry.go

```diff
@@ -142,28 +148,30 @@ func parseMirrorRegistries(
 	mirrorTrees []*toml.Tree,
 	insecureRegistries *[]string,
 	source string,
-) {
+) error {
 	itmsMirror := configv1.ImageTagMirrors{}
 	idmsMirror := configv1.ImageDigestMirrors{}
 	for _, mirrorTree := range mirrorTrees {
-		if mirror, ok := mirrorTree.Get("location").(string); ok {
-			if insecure, ok := mirrorTree.Get("insecure").(bool); ok && insecure {
-				*insecureRegistries = append(*insecureRegistries, mirror)
-			}
-			if pullFrom, ok := mirrorTree.Get("pull-from-mirror").(string); ok {
-				switch pullFrom {
-				case "tag-only":
-					itmsMirror.Source = source
-					itmsMirror.Mirrors = append(itmsMirror.Mirrors, configv1.ImageMirror(mirror))
-				case "digest-only":
-					idmsMirror.Source = source
-					idmsMirror.Mirrors = append(idmsMirror.Mirrors, configv1.ImageMirror(mirror))
-				}
-			} else {
-				// Default is pulling by digest
+		mirror, ok := mirrorTree.Get("location").(string)
+		if !ok {
+			return fmt.Errorf("missing mirror registry location key or the value of location is not a string")
+		}
+		if insecure, ok := mirrorTree.Get("insecure").(bool); ok && insecure {
+			*insecureRegistries = append(*insecureRegistries, mirror)
+		}
+		if pullFrom, ok := mirrorTree.Get("pull-from-mirror").(string); ok {
+			switch pullFrom {
+			case "tag-only":
+				itmsMirror.Source = source
+				itmsMirror.Mirrors = append(itmsMirror.Mirrors, configv1.ImageMirror(mirror))
+			case "digest-only":
 				idmsMirror.Source = source
 				idmsMirror.Mirrors = append(idmsMirror.Mirrors, configv1.ImageMirror(mirror))
 			}
+		} else {
+			// Default is pulling by digest
+			idmsMirror.Source = source
+			idmsMirror.Mirrors = append(idmsMirror.Mirrors, configv1.ImageMirror(mirror))
```

Comment:
Can we extract this to a function? Doing so we'll also get rid of the `else` condition.

Reply:
Modified, how does it look?